kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplayEngineMetricsSuite.scala

```diff
@@ -342,6 +379,271 @@ class LogReplayEngineMetricsSuite extends QueryTest
       }
     }
   }
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  // Tests for loading P & M through checksums files                                            //
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+
+  Seq(-1L, 3L, 4L).foreach { version => // -1 means latest version
+    test(s"checksum found at the read version: ${if (version == -1) "latest" else version}") {
+      withTempDirAndMetricsEngine { (path, engine) =>
+        buildTableWithCrc(path)
+        val table = Table.forPath(engine, path)
+
+        loadPandMCheckMetrics(
+          version match {
+            case -1 => table.getLatestSnapshot(engine).getSchema(engine)
+            case ver => table.getSnapshotAsOfVersion(engine, ver).getSchema(engine)
+          },
+          engine,
+          // shouldn't need to read commit or checkpoint files as P&M are found through checksum
+          expJsonVersionsRead = Nil,
+          expParquetVersionsRead = Nil,
+          expParquetReadSetSizes = Nil,
+          expChecksumReadSet = Seq(if (version == -1) 11 else version))
+      }
+    }
+  }
+
+  test("checksum not found at the read version, but found at a previous version") {
```

Comment:
I think you're missing cases for:
1. snapshot hint found for read verson AND crc found at read version ==> use hint
2. crc found at read version AND checkpoint at read version => use crc

Reply:
added both